firefox_puppeteer/ui/tabbar.py

```diff
@@ -236,6 +237,40 @@ def tab_element(self):
         """
         return self._tab_element
 
+    # Properties for backend values
+
+    @property
+    def location(self):
+        """Returns the current URL
+
+        :returns: Current URL
+        """
+        self.switch_to()
+
+        with self.marionette.using_context(self.marionette.CONTEXT_CONTENT):
+            return self.marionette.get_url()
+
+    @property
+    def certificate(self):
+        """The SSL certificate assiciated with the loaded web page.
+
+        :returns: Certificate details as JSON blob.
+        """
+        self.switch_to()
+
+        cert = self.marionette.execute_script("""
+          var securityUI = arguments[0].linkedBrowser.securityUI;
+          var status = securityUI.QueryInterface(Ci.nsISSLStatusProvider)
+                                 .SSLStatus;
+
+          return status ? status.serverCert : null;
+        """, script_args=[self.tab_element])
```

Comment:
>> +        cert = self.marionette.execute_script("""
>> +          var securityUI = arguments[0].linkedBrowser.securityUI;
>> +          var status = securityUI.QueryInterface(Ci.nsISSLStatusProvider)
>> +                                 .SSLStatus;
>> +
>> +          return status ? status.serverCert : null;
>> +        """, script_args=[self.tab_element])
> 
> Can't you do this in the security module by using gBrowser.selectedBrowser ?

We could but I don't want to force that the tab has to be selected
first. It should perfectly be doable to get the certificate for a loaded
page in a background tab.